src/app/storage/query/index.ts

```diff
@@ -0,0 +1,165 @@
+import { NextFunction, Request, Response, Router } from 'express';
+
+import createHashString from '@/utils/createHashString';
+
+import { endpointRequiresQueryId, populateResponseLocals } from './middlewares';
+import {
+  compareDifferentQueries,
+  getStoredQueryById,
+  initialiseQueryStorage,
+  validateNextQuery,
+} from './utils';
+
+const queryRouter: Router = Router({
+  caseSensitive: true,
+});
+
+queryRouter.use([endpointRequiresQueryId, populateResponseLocals]);
+
+// WIP this can be better split into separate routes (with/out queryId)
+const route = queryRouter.route('/query/:queryId?');
+
+/************************************************************************************
+ *         Allows users to store, edit and delete their own "search queries"
+ ************************************************************************************
+ *
+ *  GET /query (gets a list of all queries for the current user)
+ *  GET /query/{id} (gets a specific query by its ID)
+ *  POST /query (stores a query, and responds with the corresponding ID)
+ *  PUT /query/{id} (update if exist) idempotent
+ *  DELETE /query/{id}
+ *  DELETE /query/all (delete all)
+ *
+ ***********************************************************************************/
+
+route.get(async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const { storedQueryHandler, queryId } = res.locals;
+
+    const data = await (queryId ? getStoredQueryById(queryId) : storedQueryHandler());
+    const countData = Object.keys(data.queries).length;
+
+    return res.status(200).send({
+      data,
+      message: `Found ${countData === 1 ? 'a' : countData || 'no'} quer${
+        countData === 1 ? 'y' : 'ies'
+      }`,
+    });
+  } catch (error: any) {
+    console.error(error.message);
+    next(error);
+  }
+});
+
+route.post(async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
I've split the different methods into different blocks (rather than chain them) as per PR feedback.
bonus: added argument types